mac/deploy_mac.sh

```diff
@@ -47,15 +61,30 @@ build_app()
     local job_count
     job_count=$(sysctl -n hw.ncpu)
 
+    # Get Jamulus version
+    local app_version="$(cat "${project_path}" | sed -nE 's/^VERSION *= *(.*)$/\1/p')"
+
     make -f "${build_path}/Makefile" -C "${build_path}" -j "${job_count}"
 
     # Add Qt deployment dependencies
-    if [[ -z "$cert_name" ]]; then
+    if [[ -z "$macapp_cert_name" ]]; then
         macdeployqt "${build_path}/${target_name}.app" -verbose=2 -always-overwrite
     else
-        macdeployqt "${build_path}/${target_name}.app" -verbose=2 -always-overwrite -hardened-runtime -timestamp -appstore-compliant -sign-for-notarization="${cert_name}"
+        macdeployqt "${build_path}/${target_name}.app" -verbose=2 -always-overwrite -hardened-runtime -timestamp -appstore-compliant -sign-for-notarization="${macapp_cert_name}"
```

Comment:
All of this will break the existing ad-hoc signing and notarizing.

I think we need a second step here that signs a separate artifact for the Mac app store.

Reply:
Ok, I think I snarled things up a bit by renaming things unnecessarily.

Currently my build process produces both pkg and dmg files, and I assume the dmg file will work fine with the notarization process for adhoc release (I haven't tested that obviously as notarization is only for non app store distribution).

As far as I understand it, the certificate that you are currently using for codesigning for adhoc release can be used also for the codesigning for app-store release.
App Store distribution just requires the additional 1 certificate for use by `productbuild` - described as following in Apple Developer web UI:

  **Mac Installer Distribution
  This certificate is used to sign your app's Installer Package for submission to the Mac App Store.**

So: let me fix the cert and variable renaming so it's a transparent change, and I'll look at automating the validate and upload stages with `xcrun altool` as I mentioned in the iOS PR.

Reply:
Ok scratch that :) you are probably using the following cert type for your current code-signing:

**Developer ID Application
This certificate is used to code sign your app for distribution outside of the Mac App Store.**

In which case indeed we do need a separate certificate for codesigning for App Store release.

Ok so I will rework by adding 2 x additional certs in parallel, to not disturb the current adhoc release workflow.

Reply:
I hope this is fixed.